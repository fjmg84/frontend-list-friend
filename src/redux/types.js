export const TYPES = {
  LOAD: "load/friends",
  ERROR_ACTIVE: "error/active",
};
