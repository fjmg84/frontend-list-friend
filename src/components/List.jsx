import React from "react";
import { useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";
import image from "../photo.jpg";

export const List = () => {
  const friends = useSelector((state) => state.friendsReducer) || [];
  const navigate = useNavigate();

  return (
    <ul>
      {friends.map((friend) => {
        return (
          <li key={friend.id}>
            <div className="row">
              <div className="card">
                <img
                  src={image}
                  alt={friend.img}
                  className="rectangle1 avatar"
                />
                <span
                  className={`availability eclipse ${
                    friend.available ? "available" : "not-available"
                  }`}
                ></span>
                <div className="frame">
                  <div className="name-friend">
                    <span>{`${friend.first_name} ${friend.last_name}`}</span>
                  </div>
                  <div className="status-friend">
                    <span>{friend.status}</span>
                  </div>
                </div>
                {friend.available ? (
                  <button onClick={() => navigate(`./show/${friend.id}`)}>
                    Details
                  </button>
                ) : (
                  <button disabled>Details</button>
                )}
              </div>
            </div>
          </li>
        );
      })}
    </ul>
  );
};
