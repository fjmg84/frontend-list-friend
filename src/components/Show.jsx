import React, { useEffect, useState } from "react";
import { useParams, useNavigate } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { findOneFriend } from "../redux/actions/friends";
import { setAlert } from "../redux/actions/alerts";
import img from "../photo.jpg";

export const Show = () => {
  let { id } = useParams();
  const dispatch = useDispatch();
  const { status, msg } = useSelector((state) => state.alertsReducer);
  const friends = useSelector((state) => state.friendsReducer);
  const [statusOfTheFriend, setStatusOfTheFriend] = useState(() => {
    return friends?.filter((f) => f.id === +id)[0].status;
  });

  const [view, setView] = useState(true);
  const [friend, setFriend] = useState({});
  const bio = "";
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(setAlert(true, "loading..."));
    findOneFriend(id)
      .then((friend) => {
        setFriend(friend);
        dispatch(setAlert(false, undefined));
      })
      .catch((err) => {
        console.error(err);
        dispatch(setAlert(true, "Up!!! Sorry. Not Found data of this friend."));
      });
  }, [id, dispatch]);

  const handleView = () => {
    setView(!view);
  };

  const info = () => {
    return (
      <>
        <div className="article col">
          <p className="title">Bio:</p>
          <div className="text">{friend.bio}</div>
        </div>
        <div className="divider"></div>

        <div className="article row">
          <p className="title">Phone:</p>
          <div className="">{friend.phone}</div>
        </div>
        <div className="divider"></div>

        <div>
          <div className="article row">
            <p className="title">Address</p>
            <div>{friend.address_1}</div>
          </div>

          <div className="article row">
            <p className="title">City</p>
            <div>{friend.city}</div>
          </div>

          <div className="article row">
            <p className="title">Zipcode</p>
            <div>{friend.zipcode}</div>
          </div>

          <div className="article row">
            <p className="title">State</p>
            <div>{friend.state}</div>
          </div>
        </div>
      </>
    );
  };

  const photos = () => {
    return (
      <div className="article">
        <div className="gallery">
          <img src={img} />
          <img src={img} />
          <img src={img} />

          <img src={img} />
          <img src={img} />
          <img src={img} />

          <img src={img} />
          <img src={img} />
          <img src={img} />
        </div>
      </div>
    );
  };

  const handleNavegation = () => {
    navigate(-1);
  };

  return (
    <div className="box">
      <div className="arrow">
        <span className="fill vector" onClick={() => handleNavegation()}></span>
      </div>

      <div className="body">
        <div className="header">
          <h1>Friend</h1>
          {status && <div className="error">{msg}</div>}
        </div>

        <div className="main">
          <div className="show-card">
            <div className="show-card-img">
              <img src={img} alt={friend.img} className="medium" />
              <span className="availability eclipse available"></span>
            </div>
            <div className="show-card-content">
              <p>{`${friend.first_name} ${friend.last_name}`}</p>
              <div className="show-card-header-status">
                <span>{statusOfTheFriend}</span>
              </div>

              <div className="show-card-box-data">
                <div className="nav">
                  <div className={`tab1 ${view ? "active" : "inactive"}`}>
                    <span onClick={() => handleView()}>Info</span>
                    <span className={`${view ? "line" : ""}`}></span>
                  </div>
                  <div className={`tab2 ${!view ? "active" : "inactive"}`}>
                    <span onClick={() => handleView()}>Photos</span>
                    <span className={`${!view ? "line" : ""}`}></span>
                  </div>
                </div>

                <div className="section">{view ? info() : photos()}</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
