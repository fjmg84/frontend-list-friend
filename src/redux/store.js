import { createStore, combineReducers } from "redux";
import { friendsReducer } from "../reducer/friends";
import { errorsReducer } from "../reducer/errors";

const rootReducers = combineReducers({
  friendsReducer,
  errorsReducer,
});

export const store = createStore(
  rootReducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
