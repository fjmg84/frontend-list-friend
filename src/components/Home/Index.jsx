import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getFriends } from '../../redux/actions/friends'
import { setAlert } from '../../redux/actions/alerts'
import { TYPES } from '../../redux/types'
import { List } from './List'
import { Spinner } from '../Helps/Spinner'

export const Index = () => {
	const dispatch = useDispatch()
	const { status, msg, classAlert } = useSelector(
		(state) => state.alertsReducer,
	)
	const friends = useSelector((state) => state.friendsReducer)

	useEffect(() => {
		dispatch(setAlert(true, 'Loading friends...', 'loading'))
		getFriends()
			.then((friends) => {
				dispatch({ type: TYPES.LOAD, payload: friends })
				dispatch(setAlert(false, undefined))
			})
			.catch((err) => {
				console.log(err)
				dispatch(
					setAlert(true, 'Up!!! Sorry Not Found List Of Friends :(', 'error'),
				)
			})
	}, [dispatch])

	return (
		<div className="box">
			<div className="body">
				<div className="header">
					<h1>Friends</h1>
					{status && <div className={classAlert}>{msg}</div>}
				</div>

				<div className="main">
					{friends.length > 0 ? <List friends={friends} /> : <Spinner />}
				</div>
			</div>
		</div>
	)
}
