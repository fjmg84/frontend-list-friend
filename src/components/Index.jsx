import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getFriends } from "../redux/actions/friends";
import { setAlert } from "../redux/actions/alerts";
import { TYPES } from "../redux/types";
import { List } from "./List";

export const Index = () => {
  const dispatch = useDispatch();
  const { status, msg, classAlert } = useSelector(
    (state) => state.alertsReducer
  );

  useEffect(() => {
    dispatch(setAlert(true, "Loading friends...", "loading"));
    getFriends()
      .then((friends) => {
        dispatch({ type: TYPES.LOAD, payload: friends });
        dispatch(setAlert(false, undefined));
      })
      .catch((err) => {
        console.log(err);
        dispatch(setAlert(true, "Up!! Sorry. Not loading Friends :(", "error"));
      });
  }, [dispatch]);

  return (
    <div className="box">
      <div className="body">
        <div className="header">
          <h1>Friends</h1>
          {status && <div className={classAlert}>{msg}</div>}
        </div>

        <div className="main">
          <List />
        </div>
      </div>
    </div>
  );
};
