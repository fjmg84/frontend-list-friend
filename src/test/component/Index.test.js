import React from "react";
import { Index } from "../../components/Index";
import { mount, shallow } from "enzyme";
import { Provider } from "react-redux";
import { store } from "../../redux/store";
import { MemoryRouter } from "react-router-dom";

describe("testing component <Index/>", () => {
  test("shallow component <Index/>", () => {
    const wrapper = shallow(
      <MemoryRouter>
        <Provider store={store}>
          <Index />
        </Provider>
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("mount component <Index/>", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <Index />
        </Provider>
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
