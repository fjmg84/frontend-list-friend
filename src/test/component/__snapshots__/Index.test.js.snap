// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component <Index/> mount component <Index/> 1`] = `
<div
  className="body"
>
  <h1>
    Friends
  </h1>
  <div
    className="error"
  >
    Loading friends...
  </div>
  <div
    className="box"
  >
    <ul />
  </div>
</div>
`;

exports[`testing component <Index/> shallow component <Index/> 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "tgzkss44",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "index": 0,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "tgzkss44",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Index />
  </Provider>
</Router>
`;
